### PR TITLE
Fixes brain problems not equipping medicine properly.

### LINKED ISSUE #4453: Brain problems quirk can fail to equip
Issue reported from  Round ID: 30237 (BeeStation Sage - Roleplay)

Reporting client version: 514.1554

[00:03:51] Runtime in inventory.dm, line 19: list index out of bounds
proc name: equip in one of slots (/mob/living/carbon/proc/equip_in_one_of_slots)
src: (/mob/living/carbon/human)
src.loc: the floor (113,91,2) (/turf/open/floor/circuit)
call stack:
(/mob/living/carbon/human): equip in one of slots(the bottle of mannitol pills (/obj/item/storage/pill_bottle/mannitol/braintumor), /list (/list), 0)
Brain Tumor (/datum/quirk/brainproblems): on spawn()
Brain Tumor (/datum/quirk/brainproblems): New( (/mob/living/carbon/human), 1)
(/mob/living/carbon/human): add quirk(/datum/quirk/brainproblems (/datum/quirk/brainproblems), 1)
Quirks (/datum/controller/subsystem/processing/quirks): AssignQuirks( (/mob/living/carbon/human), (/client), 1)
Ticker (/datum/controller/subsystem/ticker): equip characters()
Ticker (/datum/controller/subsystem/ticker): setup()
Ticker (/datum/controller/subsystem/ticker): fire(0)
Ticker (/datum/controller/subsystem/ticker): ignite(0)

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Fixes #4453
Changes the usage of equip_in_one_of_slots to be used how it is meant to be used.

## Why It's Good For The Game

Fixes a runtime error occured by improper use of the equip_in_one_of_slots  proc.

## Testing Photographs and Procedure

image

## Changelog
:cl:
fix: Fixes brain trauma quirk medicine spawning
/:cl:

### CODE REVIEW

File: code/datums/traits/negative.dm [hunk 1]
```diff
@@ -67,17 +67,18 @@
...
+		"in your backpack" = ITEM_SLOT_BACKPACK
+	)
+	where = H.equip_in_one_of_slots(P, slots, FALSE)
 
 /datum/quirk/brainproblems/post_add()
-	to_chat(quirk_holder, "<span class='boldnotice'>There is a bottle of mannitol [where]. You're going to need it.</span>")
+	if(where)
+		to_chat(quirk_holder, "<span class='boldnotice'>There is a bottle of mannitol [where]. You're going to need it.</span>")
+	else
+		to_chat(quirk_holder, "<span class='boldnotice'>You dropped your bottle of mannitol on the shuttle ride here and couldn't find it, it must have fallen into a bluespace pocket! You better find some more at chemistry and fast...</span>")
```
Comment (Archanial) [inline code/datums/traits/negative.dm]: Spawn it under the player

**Review by Archanial — CHANGES_REQUESTED** [review]
CL needs to be a bit more detailed

File: code/datums/traits/negative.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/datums/traits/negative.dm]: it does spawn under the player actually

Comment (PowerfulBacon) [inline code/datums/traits/negative.dm]:
```suggestion
		to_chat(quirk_holder, "<span class='boldnotice'>You dropped your bottle of mannitol on the floor. better pick it up, you are going to need it.</span>")
```

File: code/datums/traits/negative.dm [hunk 2]
```diff
@@ -78,7 +78,7 @@
 	if(where)
 		to_chat(quirk_holder, "<span class='boldnotice'>There is a bottle of mannitol [where]. You're going to need it.</span>")
 	else
-		to_chat(quirk_holder, "<span class='boldnotice'>You dropped your bottle of mannitol on the shuttle ride here and couldn't find it, it must have fallen into a bluespace pocket! You better find some more at chemistry and fast...</span>")
+		to_chat(quirk_holder, "<span class='boldnotice'>You dropped your bottle of mannitol on the floor. better pick it up, you are going to need it.</span>")
```
Comment (Archanial) [inline code/datums/traits/negative.dm]:
```suggestion
		to_chat(quirk_holder, "<span class='boldnotice'>You dropped your bottle of mannitol on the floor. Better pick it up, you are going to need it.</span>")
```
🅱️

**Review by Archanial — APPROVED** [review]

**Pull request merged**